examples/embedding/main_test.go

```diff
@@ -2,44 +2,55 @@ package main
 
 import (
 	"bytes"
+	"flag"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"testing"
 )
 
-const embeddingTestOutput = "embedding-test-output.txt"
+const embeddingTestOutput = "testdata/embedding_out_golden.txt"
+
+var regen = flag.Bool("regen", false, "regenerate golden files")
 
 func TestEmbeddedExample(t *testing.T) {
 
-	cmd := exec.Command("go", "build", ".")
-	err := cmd.Run()
+	tmp, err := os.MkdirTemp("", "go-python-embedding-")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmp)
+
+	cmd := exec.Command("go", "build", "-o", filepath.Join(tmp, "exe"), ".")
```

Comment:
little windows snowflake :)
```suggestion
	exe := filepath.Join(tmp, "out.exe")
	cmd := exec.Command("go", "build", "-o", exe, ".")
```